test/test-applications/integrations/TestApplication.Http.NetFramework/TestServer.cs

```diff
@@ -0,0 +1,135 @@
+// <copyright file="TestServer.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Text;
+using System.Threading;
+using TestApplication.Http.NetFramework.Helpers;
+
+namespace TestApplication.Http.NetFramework;
+
+public class TestServer : IDisposable
+{
+    private static readonly ActivitySource MyActivitySource = new ActivitySource("TestApplication.Http.NetFramework", "1.0.0");
+
+    private readonly HttpListener _listener;
+    private readonly Thread _listenerThread;
+    private readonly string _prefix;
+
+    public TestServer(string sufix)
+    {
+        // try up to 5 consecutive ports before giving up
+        int retries = 4;
+        while (true)
+        {
+            // seems like we can't reuse a listener if it fails to start,
+            // so create a new listener each time we retry
+            _listener = new HttpListener();
+
+            try
+            {
+                _listener.Start();
+
+                Port = TcpPortProvider.GetOpenPort();
+                _prefix = new UriBuilder("http", "localhost", Port, sufix).ToString();
+                _listener.Prefixes.Add(_prefix);
+
+                // successfully listening
+                _listenerThread = new Thread(HandleHttpRequests);
+                _listenerThread.Start();
+                Console.WriteLine($"[LISTENER] Listening on '{_prefix}'");
+                return;
+            }
+            catch (HttpListenerException) when (retries > 0)
+            {
+                _listener.Close(); // a new listener is created in the beginnning of the loop
+                retries--;
+            }
+
+            _listener.Close(); // always close listener if exception is thrown, whether it was caught or not
+            throw new InvalidOperationException("Listener shut down. Could not find available port.");
```

Comment:
This seems to break any retry, no?

Reply:
you are right 😄 
we miss a `continue` in `catch`

Reply:
PTAL at https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1634